migrations/db/migrations/20230607094822_add_auth_schema_protections.sql

```
@@ -0,0 +1,84 @@
+-- migrate:up
+
+create sequence if not exists
+  auth.changes_seq increment by 1;
+
+comment on sequence auth.changes_seq is 'Tracks the number of changes of the auth schema.';
+
+grant all on sequence
+  auth.changes_seq
+  to supabase_auth_admin
+  with grant option;
+
+grant usage, select on sequence
+  auth.changes_seq
+  to supabase_admin;
+  -- supabase_admin can advance the sequence only
+
+create sequence if not exists
```

nit:
since this sequence should never exist prior to this migration, remove `if not exists`. If something has gone wrong and the sequence does exist, its better to fail loudly during a migration than to hide the issue

---

same for `changes_seq`

---

These migrations are also run when doing restores, so we need `if not exists` to make them idempotent